pywbem_mock/_inmemoryrepository.py

```diff
@@ -0,0 +1,264 @@
+
+        super(InMemoryObjectStore, self).__init__(cim_object_type)
+
+        # Define the dictionary that implements the object store.
+        # The keys in this dictionary are the names of the objects and
+        # the values the corresponding CIM objects.
+        if cim_object_type.__name__ in ("CIMClass", 'CIMQualifierDeclaration'):
+            self._data = NocaseDict()
+        elif cim_object_type.__name__ == 'CIMInstance':
+            self._data = {}
+        else:
+            assert False, "InMemoryObjectStore: Invalid input parameter {}." \
+                .format(cim_object_type)
+    # pylint: enable=line-too-long
+
+    def __repr__(self):
+        return _format('InMemoryObjectStore(type={0},  dict={1}, size={2}',
+                       self._cim_object_type, type(self._data),
+                       len(self._data))
+
+    def exists(self, name):
+        return name in self._data
+
+    def get(self, name, copy=True):
+        # pylint: disable=no-else-return
+        if name in self._data:
+            if copy:
+                return self._data[name].copy()
+            return self._data[name]
+        else:
+            raise KeyError('{} not in {} object store'
+                           .format(name, self._cim_object_type))
+
+    def create(self, name, cim_object):
+        assert isinstance(cim_object, self._cim_object_type)
+
+        if name in self._data:
+            raise ValueError('{} already in {} object store'
+                             .format(name, self._cim_object_type))
+
+        self._data[name] = cim_object.copy()
+
+    def update(self, name, cim_object):
+        assert isinstance(cim_object, self._cim_object_type)
+
+        if name not in self._data:
+            raise KeyError('{} not in {} object store'
+                           .format(name, self._cim_object_type))
+
+        # Replace the existing object with a copy of the input object
+        self._data[name] = cim_object.copy()
+
+    def delete(self, name):
+        if name in self._data:
+            del self._data[name]
+        else:
+            raise KeyError('{} not in {} object store'
+                           .format(name, self._cim_object_type))
+
+    def iter_names(self):
+        return six.iterkeys(self._data)
+
+    def iter_values(self, copy=True):
+        for value in six.itervalues(self._data):
+            if copy:
+                yield(value.copy())
+            else:
+                yield(value)
+
+    def len(self):
+        return len(self._data)
+
+
+class InMemoryRepository(BaseRepository):
+    """
+    A CIM repository that maintains its data in memory.
+
+    Documentation for the methods and properties inherited from
+    ~pywbem_mock:`BaseObjectStore` is also inherited in the pywbem
+    documentation. Therefore the methods in this class have no
+    documentation string.
+    """
+    def __init__(self, initial_namespace=None):
+        """
+        Initialize the InMemoryRepository.
```

Comment:
The docstring of `__init__()` contains "Initialize the InMemoryRepository.". If you look at the generated docs, it does not ready very nicely there. I suggest to remove that sentence and to have only the Parameters section in that docstring.

Reply:
DONE

Reply:
By the way, while it works in simple case, the merge tools does not handle merging things very well in some cases so we have to be careful.